noir-projects/noir-contracts/contracts/nft_contract/src/main.nr

```diff
@@ -134,21 +135,41 @@ contract NFT {
         public_owners_storage.write(to);
     }
 
-    /// Prepares a transfer from public balance of `from` to a private balance of `to`. The transfer then needs to be
-    /// finalized by calling `finalize_transfer_to_private`. `transient_storage_slot_randomness` is passed
-    /// as an argument so that we can derive `transfer_preparer_storage_slot_commitment` off-chain and then pass it
-    /// as an argument to the followup call to `finalize_transfer_to_private`.
+    // Transfers token with `token_id` from public balance of message sender to a private balance of `to`.
     #[private]
-    fn prepare_transfer_to_private(
-        from: AztecAddress,
-        to: AztecAddress,
-        transient_storage_slot_randomness: Field
-    ) {
+    fn transfer_to_private(to: AztecAddress, token_id: Field) {
+        let from = context.msg_sender();
+
+        let nft = NFT::at(context.this_address());
+
+        // We prepare the transfer.
+        let hiding_point_slot = nft._prepare_transfer_to_private(to).call(&mut context);
+
+        // At last we finalize the transfer. Usafe of the `unsafe` method here is safe because we set the `from`
+        // function argument to a message sender, guaranteeing that he can transfer only his own NFTs.
+        nft._finalize_transfer_to_private_unsafe(from, token_id, hiding_point_slot).enqueue(&mut context);
+    }
+
+    /// Prepares a transfer to a private balance of `to`. The transfer then needs to be
+    /// finalized by calling `finalize_transfer_to_private`. Returns a hiding point slot.
+    #[private]
+    fn prepare_transfer_to_private(to: AztecAddress) -> Field {
+        NFT::at(context.this_address())._prepare_transfer_to_private(to).call(&mut context)
+    }
+
+    /// This function exists separately from `prepare_transfer_to_private` solely as an optimization as it allows
+    /// us to have it inlined in the `transfer_to_private` function which results in one less kernel iteration.
+    ///
+    /// TODO(#9180): Consider adding macro support for functions callable both as an entrypoint and as an internal
+    /// function.
+    #[private]
+    #[internal]
```

Comment:
It's not really inlined though is it? Should this not be a `#[contract_library_function]`?

Reply:
Good catch. A stupid mistake. Addressed in https://github.com/AztecProtocol/aztec-packages/pull/9150/commits/121a7533ad36ee98233bc036ba23d64dc1b32ea1